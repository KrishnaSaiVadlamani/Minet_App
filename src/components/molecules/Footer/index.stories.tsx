import { ComponentMeta, ComponentStory } from '@storybook/react'
import React from 'react'
import Footer from './index'

export default {
  title: 'Molecules/Footer',
  component: Footer,
} as ComponentMeta<typeof Footer>

const Template: ComponentStory<typeof Footer> = () => (
  <Footer/>
)

export const SimpleFooter = Template.bind({})
