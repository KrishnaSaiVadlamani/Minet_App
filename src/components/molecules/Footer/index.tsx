

import { Grid, styled, Stack, Box } from '@mui/material'
import React from 'react'
import theme from '../../../theme/index'
import ButtonComponent from '../../atoms/Button'
import TypographyComponent from '../../atoms/Typography'
import DropDownComponent from '../Dropdown'

const StyledBox = styled(Box)({
  [theme.breakpoints.up('lg')]:{
    maxWidth: '1500px'
  },
  [theme.breakpoints.up('xl')]: {
    maxWidth: '98vw'
  },
  [theme.breakpoints.down('lg')]: {
    minWidth: '1366px'
  },
  [theme.breakpoints.between('lg', 'xl')]: {
    width: '97vw',
  },
 
  paddingLeft: '104px',
})

const MainContainer = styled(Grid)({
  height: '90px',
  borderTop: `1px solid ${theme.palette.greyColors.grey100}`,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  paddingRight: '22px'
})

const StyledTypographyBlue = styled(TypographyComponent)({
  color: theme.palette.primary.primary500,
})

const StyledTypographyBlack = styled(TypographyComponent)({
  color: theme.palette.textColor.highEmphasis,
})

const StyledButton = styled(ButtonComponent)({
  width: '109px',
  height: '42px',
  border: `1px solid ${theme.palette.primary.main}`,
  fontWeight: 'bold',
  '&:hover': {
    backgroundColor: theme.palette.structural.main,
  },
})

const RightContainer = styled(Grid)({
  alignItems: 'flex-end',
  margin: '0px',
})

const Footer = () => {
  return (
    <StyledBox data-testid='footer'>
      <MainContainer container>
        <Grid item xs={6}>
          <Stack direction="row" spacing={3}>
            <StyledTypographyBlue variant="body2" children={'Dashboard'} />
            <StyledTypographyBlue variant="body2" children={'Careers'} />
            <StyledTypographyBlue
              variant="body2"
              children={'Legal & Privacy'}
            />
            <StyledTypographyBlack variant="body2" children={'Â© 2021 Minet'} />
          </Stack>
        </Grid>
        <RightContainer item xs={6} justifyContent="flex-end">
          <Stack direction="row" spacing={2} justifyContent="flex-end">
            <DropDownComponent onChange={() => {}} menuList={['English']} width={'170px'}/>
            <StyledButton variant="outlined" children="NEED HELP" backgroundColor={'none'}/>
          </Stack>
        </RightContainer>
      </MainContainer>
    </StyledBox>
  )
}

export default Footer
