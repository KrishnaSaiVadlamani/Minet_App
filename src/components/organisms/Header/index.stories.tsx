import React from "react"
import Header from "."
import { ComponentStory } from '@storybook/react'


export default{
    title: "organisms/Header",
    component: Header,
}

const Template: ComponentStory<typeof Header> = (args) => <Header {...args}/>

export const Trade = Template.bind({})

Trade.args={
    pageName: 'Trade',
}