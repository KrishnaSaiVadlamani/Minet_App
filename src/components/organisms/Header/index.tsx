import { Box, Divider, Grid, styled } from "@mui/material";
import React from "react";
import theme from "../../../theme";
import ButtonComponent from "../../atoms/Button";
import TypographyComponent from "../../atoms/Typography";
import AvatarDropdown from "../../molecules/AvatarDropdown";

interface IHeaderProps {
  pageName: string;
}

const CustomBox = styled(Box)({
  padding: "20px 24px 20px 24px",
  [theme.breakpoints.up("lg")]: {
    maxWidth: "1500px",
  },
  [theme.breakpoints.up("xl")]: {
    maxWidth: "98vw",
  },
  [theme.breakpoints.down("lg")]: {
    minWidth: "1366px",
  },
  [theme.breakpoints.between("lg", "xl")]: {
    width: "97vw",
  },
  borderBottom: `1px solid ${theme.palette.greyColors.grey100}`,
});
const CustomButton = styled(ButtonComponent)({
  borderRadius: "4px",
  padding: "0px 16px 0px 16px",
  width: "120px",
  height: "42px",
});

const StyledTypographyComponent = styled(TypographyComponent)({
  paddingTop: "4.5px",
});

const Header = (props: IHeaderProps) => {
  const { pageName } = props;
  return (
    <CustomBox>
      <Grid container>
        <Grid item xs={6}>
          <StyledTypographyComponent
            variant="h6"
            width="122px"
            height="34px"
            color="textColor.highEmphasis"
          >
            {pageName}
          </StyledTypographyComponent>
        </Grid>

        <Grid
          item
          container
          wrap="nowrap"
          xs={6}
          justifyContent="flex-end"
          columnGap={2}
        >
            <Grid item>
                <CustomButton
                  variant="contained" label={"SELL"}
                 sx = {{backgroundColor: theme.palette.primary.warning300}}
                >
                  <TypographyComponent variant="button">
                    SELL
                  </TypographyComponent>
                </CustomButton>
            </Grid>

          <Grid item>
              <CustomButton variant="contained" sx = {{backgroundColor:"primary"}} label={"BUY"}>
                <TypographyComponent variant="button">BUY</TypographyComponent>
              </CustomButton>
          </Grid>
          <Divider orientation="vertical" variant="middle" flexItem />
          <Grid item display="flex" alignItems="center">
            <AvatarDropdown />
          </Grid>
        </Grid>
      </Grid>
    </CustomBox>
  );
};

export default Header;
