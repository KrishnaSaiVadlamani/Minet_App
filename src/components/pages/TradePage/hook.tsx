import { useState } from "react"
import { timeProps } from "../../../utils/types"

export const useTimePeriodHook = () => {
    const [timePeriod, setTimePeriod] = useState<timeProps>('1h')
    const [searchValue, setSearchValue] = useState<string>("")

    const handleTimeDropDownChange = (value: string) => {
        let time = value
        const len = time.length - 1
        time = time.substring(0,len) + time.charAt(len).toLowerCase();
        setTimePeriod(time as timeProps)
    }
    const handleSearchFieldChange = (value: string) => {
        setSearchValue(value)
      }

    return {timePeriod, setTimePeriod, searchValue, setSearchValue, handleTimeDropDownChange, handleSearchFieldChange}
}