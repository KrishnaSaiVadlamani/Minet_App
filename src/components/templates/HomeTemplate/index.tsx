import { Box, Grid, styled } from "@mui/material";
import React from "react";
import theme from "../../../theme"

interface HomeTemplateProps {
  header: React.ReactNode
  sideNav: React.ReactNode
  footer: React.ReactNode
  children: React.ReactNode
  paddingTop?: string
}

const StyledChildren = styled(Grid)({
  paddingLeft: "24px",
  borderLeft: `1px solid ${theme.palette.greyColors.grey100}`,
  backgroundColor: theme.palette.greyColors.grey400,
});

const StyledHeader = styled(Grid)({
  height: "83px",
});

const RightContainer = styled(Grid)({
  paddingLeft: "80px",
});

const StyledSideNav = styled(Box)({
  borderRight: `1px solid ${theme.palette.greyColors.grey100}`,
});

const HomeTemplate: React.FC<HomeTemplateProps> = ({
    header, 
    sideNav, 
    footer, 
    children,
    paddingTop
}) => {
     return(
        <Box data-testid="home-template">
            <StyledSideNav>{sideNav}</StyledSideNav>
            <RightContainer container direction="column">
                <StyledHeader item>{header}</StyledHeader>
                <StyledChildren item paddingTop={paddingTop? paddingTop: '0px'}>{children}</StyledChildren>
            </RightContainer>
            {footer}
        </Box>
     )
}

export default HomeTemplate