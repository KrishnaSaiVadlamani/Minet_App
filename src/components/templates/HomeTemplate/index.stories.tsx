import { ComponentStory } from "@storybook/react";
import React from "react";
import HomeTemplate from ".";
import Header from "../../organisms/Header";
import Footer from '../../molecules/Footer'
import SideNavCompnent from '../../organisms/SideNav'

export default {
  title: "templates/HomeTemplate",
  component: HomeTemplate,
};

const Template: ComponentStory<typeof HomeTemplate> = (args) => (
  <HomeTemplate {...args} />
);

export const SimpleHomeTemplate = Template.bind({})

SimpleHomeTemplate.args = {
    header: <Header pageName='Trade' />,
    footer: <Footer />,
    sideNav: <SideNavCompnent />,
    children: <div style={{ height: '896px', width: '1296px' }}>content</div>
}