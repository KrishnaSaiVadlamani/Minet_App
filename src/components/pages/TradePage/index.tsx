import { Box, Stack, styled, Tab, Tabs, Grid } from "@mui/material";
import DropDownComponent from "../../molecules/Dropdown";
import Footer from "../../molecules/Footer";
import SearchField from "../../molecules/Searchfield";
import Header from "../../organisms/Header";
import SideNavCompnent from "../../organisms/SideNav";
import HomeTemplate from "../../templates/HomeTemplate";
import { timeValues } from "../../../utils/constants";
import TradeTable from "../../organisms/TradeTable";
import theme from "../../../theme";
import { useTimePeriodHook } from "./hook";
import { timeProps } from "../../../utils/types";
import React, { useState } from "react";
import { TabPanelProps } from "../../organisms/Tabs";

const SearchFieldBox = styled(Box)({
  display: "flex",
  justifyContent: "flex-end",
  marginBottom: "12px",
  paddingRight: "23px",
});

const TableBox = styled(Box)({
  paddingRight: "23px",
  paddingBottom: "30px",
  minHeight: "650px",
});

const StyledBox = styled(Box)({
  paddingTop: "24px",
});

const StyledTabs = styled(Tabs)({
  "& .MuiButtonBase-root": {
    padding: "0px",
    minHeight: "28px",
  },
  "& .MuiTabs-indicator": {
    top: "35px",
  },
  "& .MuiTabs-root": {
    padding: "0px",
  },
  minHeight: "37px !important",
  width: "100%",
  borderBottom: `1px solid ${theme.palette.greyColors.grey100}`,
});

function a11yProps(index: number) {
  return {
    id: `simple-tab-${index}`,
    "aria-controls": `simple-tabpanel-${index}`,
  };
}

function TabPanel(props: TabPanelProps) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && <Box>{children}</Box>}
    </div>
  );
}

const TradePage = () => {
  const [value, setValue] = useState(0);
  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setValue(newValue);
  };
  const {
    timePeriod,
    searchValue,
    handleSearchFieldChange,
    handleTimeDropDownChange,
  } = useTimePeriodHook();

  const renderSearchFields = () => {
    return (
      <Grid container justifyContent={"space-between"}>
        <Grid item xs={8}>
          <StyledTabs value={value} onChange={handleChange}>
            <Tab label="All Assests" {...a11yProps(0)} defaultChecked={true} />
            <Tab label="WatchList" {...a11yProps(1)} />
          </StyledTabs>
        </Grid>
        <Grid item>
          <SearchFieldBox>
            <Stack spacing={"12px"} direction="row">
              <SearchField
                placeholder="Search all assets"
                filter={false}
                handleChange={handleSearchFieldChange}
                backgroundColor={theme.palette.structural.main}
                value={searchValue}
              />
              <DropDownComponent
                width={"78px"}
                menuList={timeValues}
                onChange={handleTimeDropDownChange}
                backgroundColor={theme.palette.structural.main}
              />
              <DropDownComponent
                width={"122px"}
                menuList={["All assets"]}
                onChange={() => {}}
                backgroundColor={theme.palette.structural.main}
              />
            </Stack>
          </SearchFieldBox>
        </Grid>
      </Grid>
    );
  };

  const renderChildren = (timePeriod: timeProps) => {
    return (
      <StyledBox>
        {renderSearchFields()}
        {renderTable(timePeriod)}
      </StyledBox>
    );
  };

  const renderTable = (timePeriod: timeProps) => {
    return (
      <>
        <TabPanel value={value} index={0}>
          <TableBox>
            <TradeTable timePeriod={timePeriod} filterValue={searchValue} />
          </TableBox>
        </TabPanel>
        <TabPanel value={value} index={1}>
          <TableBox>
            <TradeTable timePeriod={timePeriod} filterValue={searchValue} tableType={"watchlist"} />
          </TableBox>
        </TabPanel>
      </>
    );
  };

  return (
    <HomeTemplate
      header={<Header pageName={"Trade"} />}
      footer={<Footer />}
      sideNav={<SideNavCompnent />}
      children={renderChildren(timePeriod)}
    />
  );
};

export default TradePage;
