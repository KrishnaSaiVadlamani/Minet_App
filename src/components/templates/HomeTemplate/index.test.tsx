import renderer from "react-test-renderer";
import HomeTemplate from ".";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";
import Header from "../../organisms/Header";
import Footer from "../../molecules/Footer";
import SideNavComponentLocation from "../../organisms/SideNav";
import React from "react";
import DetailPage from "./index";
import TradePage from "../../pages/TradePage";

it("renders home template correctly", () => {
  render(
    <Router>
      <HomeTemplate
        children={<TradePage />}
        header={<Header pageName={"Trade"} />}
        footer={<Footer />}
        sideNav={<SideNavComponentLocation />}
      />
    </Router>
  );
});

describe("Templates/Home Template", () => {
  test("Renders home template correctly", () => {
    render(
      <Router>
        <HomeTemplate
          header={<Header pageName="TradePage" />}
          footer={<Footer />}
          children={
            <div style={{ height: "896px", width: "1296px" }}>content</div>
          }
          sideNav={<SideNavComponentLocation />}
        />
      </Router>
    );
    const template = screen.getByTestId("home-template");
    expect(template).toBeInTheDocument();
  });
});
