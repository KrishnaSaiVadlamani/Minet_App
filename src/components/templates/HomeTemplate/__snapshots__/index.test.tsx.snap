// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home template correctly 1`] = `
<div
  className="MuiBox-root css-0"
  data-testid="home-template"
>
  <div
    className="MuiBox-root css-waw6ev"
  >
    <div
      className="MuiGrid-root css-1gzqqop-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-laq2i1"
      >
        <img
          data-testid="image"
          height="32px"
          src={
            {
              "process": [Function],
            }
          }
          width="32px"
        />
      </div>
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="MuiBox-root css-laq2i1"
        >
          <div>
            <img
              alt="icon"
              className="css-1uu7u3w"
              data-testid="iconComponent"
              height="20px"
              src={
                {
                  "process": [Function],
                }
              }
              width="20px"
            />
          </div>
        </div>
      </a>
      <div
        className="MuiBox-root css-laq2i1"
      >
        <div>
          <img
            alt="icon"
            className="css-1mj0ve"
            data-testid="iconComponent"
            height="20px"
            src={
              {
                "process": [Function],
              }
            }
            width="auto"
          />
        </div>
      </div>
      <div
        className="MuiBox-root css-laq2i1"
      >
        <div>
          <img
            alt="icon"
            className="css-1mj0ve"
            data-testid="iconComponent"
            height="20px"
            src={
              {
                "process": [Function],
              }
            }
            width="auto"
          />
        </div>
      </div>
      <div
        className="MuiBox-root css-laq2i1"
      >
        <div>
          <img
            alt="icon"
            className="css-1mj0ve"
            data-testid="iconComponent"
            height="20px"
            src={
              {
                "process": [Function],
              }
            }
            width="auto"
          />
        </div>
      </div>
      <div
        className="MuiBox-root css-laq2i1"
      >
        <div>
          <img
            alt="icon"
            className="css-1mj0ve"
            data-testid="iconComponent"
            height="20px"
            src={
              {
                "process": [Function],
              }
            }
            width="auto"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-6r5xil-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-4rmxrd-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-138dc1c"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <h6
              className="MuiTypography-root MuiTypography-h6 css-1xhej88-MuiTypography-root"
              data-testid="Hello"
            >
              TradePage
            </h6>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-6 css-19s4pri-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1retnad-MuiButtonBase-root-MuiButton-root"
                data-testid="button"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiTypography-root MuiTypography-button css-11s3blp-MuiTypography-root"
                  data-testid="Hello"
                >
                  SELL
                </span>
              </button>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-nmgn3f-MuiButtonBase-root-MuiButton-root"
                data-testid="button"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiTypography-root MuiTypography-button css-11s3blp-MuiTypography-root"
                  data-testid="Hello"
                >
                  BUY
                </span>
              </button>
            </div>
            <hr
              className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-16bghm4-MuiDivider-root"
            />
            <div
              className="MuiGrid-root MuiGrid-item css-ev762u-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container css-ljvmaf-MuiGrid-root"
              >
                <div
                  className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div>
                    <div
                      className="MuiAvatar-root MuiAvatar-circular css-1azxqog-MuiAvatar-root"
                      data-testid="avatar"
                    >
                      <img
                        alt="Avatar"
                        className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                        src={
                          {
                            "process": [Function],
                          }
                        }
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item css-hhlda7-MuiGrid-root"
                >
                  <div>
                    <img
                      alt="icon"
                      className="css-12z3ws7"
                      data-testid="iconComponent"
                      height="13px"
                      src={
                        {
                          "process": [Function],
                        }
                      }
                      width="13px"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-12fsxbk-MuiGrid-root"
    >
      <div
        style={
          {
            "height": "896px",
            "width": "1296px",
          }
        }
      >
        content
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-vhpt6z"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-1xut4gs-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="css-u10pvi-MuiStack-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-9syhwr-MuiTypography-root"
            data-testid="Hello"
          >
            Dashboard
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-9syhwr-MuiTypography-root"
            data-testid="Hello"
          >
            Careers
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-9syhwr-MuiTypography-root"
            data-testid="Hello"
          >
            Legal & Privacy
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-1iykimn-MuiTypography-root"
            data-testid="Hello"
          >
            Â© 2021 Minet
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-k97rmi-MuiGrid-root"
      >
        <div
          className="css-1lnp4wh-MuiStack-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-1oufaa8-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <p
                className="MuiTypography-root MuiTypography-body1 css-4umqm8-MuiTypography-root"
                data-testid="Hello"
              >
                English
              </p>
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              tabIndex={-1}
              value="English"
            />
            <div
              className="MuiBox-root css-17t8bxi"
            >
              <div>
                <img
                  alt="icon"
                  className="css-12z3ws7"
                  data-testid="iconComponent"
                  height="13px"
                  src={
                    {
                      "process": [Function],
                    }
                  }
                  width="13px"
                />
              </div>
            </div>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-ihdtdm"
              >
                <span
                  className="notranslate"
                >
                  â
                </span>
              </legend>
            </fieldset>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-b63ts3-MuiButtonBase-root-MuiButton-root"
            data-testid="button"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            NEED HELP
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
